### ISSUE TITLE
Handler hangs when processing upgrade request 

### ISSUE THREAD

**Issue by khvzak** [ntex-rs/ntex#185]
ntex latest (v0.6.3), `features = ["tokio"]`

Example to reproduce:
```rust
async fn test_service(req: ntex::web::HttpRequest) -> String {
    if req.head().upgrade() {
        println!("got upgrade request");
    }
    let res =
        ntex::rt::spawn_blocking(|| std::net::ToSocketAddrs::to_socket_addrs("rust-lang.com:443"))
            .await
            .unwrap();
    println!("resolved to {res:?}");
    "Ok".into()
}

#[ntex::main]
async fn main() -> Result<(), std::io::Error> {
    std::env::set_var("RUST_LOG", "trace");
    env_logger::init();

    ntex::web::HttpServer::new(|| {
        ntex::web::App::new().default_service(ntex::web::to(test_service))
    })
    .bind("127.0.0.1:8080")?
    .run()
    .await
}
```

then we can use curl to send the following request:
```
$ curl http://127.0.0.1:8080/ -H 'sec-websocket-key: gOVHFRVBchMtb2eaAKdPLw==' -H 'connection: upgrade' -H 'sec-websocket-version: 13' -H 'upgrade: websocket' -v
```
And the handler would hang after the message `got upgrade request`. This does not happen when send a normal request.

**Comment by fafhrd91** [comment]
interesting. i wont be able to check until weekend, maybe next week. but something is definitely not right

**Comment by fafhrd91** [comment]
this problem is fixed in master. but you have to use web::ws::start or write response directly to io, h1 dispatcher cannot do anything when handler is in ws upgrade mode.

**Comment by khvzak** [comment]
Thanks @fafhrd91 for fixing this!

There is an edge case, although. It's impossible to reject an upgrade request by sending non-101 response with a body. 
I'm getting error `Stream responses are not supported for upgrade requests`. 
Basically this error would appear for any normal (non-websocket) handlers when sending upgrade requests.
Is this expected?

**Comment by fafhrd91** [comment]
i think this is bug, h1 dispatcher should not try write response. i will check

**Comment by khvzak** [comment]
Maybe I'm wrong, but should dispatcher continue working as usual for all types of requests (upgrade and non-upgrade) when underlying IO wasn't taken? 

`ws::start`  as well as low-level websocket handling requires taking IO object off which makes impossible returning response from handler.

**Comment by fafhrd91** [comment]
should be fixed in 0.6.5
